integration-tests/src/test/java/eu/tsystems/mms/tic/testframework/test/pagefactory/CheckPageTest.java

```diff
@@ -82,4 +97,38 @@ public void testT09_checkCheckRule_IsNotPresent() {
         PAGE_FACTORY.createPage(PageWithCheckRuleIsNotPresent.class, getClassExclusiveWebDriver());
     }
 
+    @Test
+    public void testT10_checkOptionalElement_IsNotPresent() {
+        PAGE_FACTORY.createPage(PageWithOptionalElement.class, getClassExclusiveWebDriver());
+    }
+
+    @Test(priority = 1)
+    @Fails(description = "The test itself passes, but collected assertions will always fail")
+    public void testT11_checkCollectedElements_IsNotPresent() {
+        PAGE_FACTORY.createPage(PageWithCollectedElement.class, getClassExclusiveWebDriver());
+    }
+
+    @Test(priority = 2)
+    public void testT11a_checkCollectedElements_verifyResult() {
+        IExecutionContextController instance = Testerra.getInjector().getInstance(IExecutionContextController.class);
+        Optional<Stream<ErrorContext>> errorContextStream = ((ClassContext) instance.getCurrentMethodContext().get().getParentContext())
```

Comment:
```java
final var  errorContextStream = instance.getCurrentMethodContext().stream()
  .map(AbstractContext::getParentContext)
  .map(ClassContext.class::cast)
  .flatMap(ClassContext::readMethodContexts)
  .filter(context -> "testT11_checkCollectedElements_IsNotPresent".equals(context.getName())) // Yoda Style - not necessary here, but a nice habit in general
  .findAny() // presumption: There can be only one. findFirst, if order matters
  .map(MethodContext::readErrors);
```
?

Note: If the assertTrue below yields important information, it should probably be `Assert#assertTrue(boolean, java.lang.String)` with an explanation. (At least I would require one at this point.)
If it does not, it may be inlined into the mapping, eliminating `errorContextStream` altogether:
```java
final var errorList = instance.getCurrentMethodContext().stream()
  .map(AbstractContext::getParentContext)
  .map(ClassContext.class::cast)
  .flatMap(ClassContext::readMethodContexts)
  .filter(context -> "testT11_checkCollectedElements_IsNotPresent".equals(context.getName()))
  .flatMap(MethodContext::readErrors) // presumption: There can be only one. insert findFirst().stream() before this flatMap, if order matters.
  .map(context -> context.getThrowable().getMessage())
  .collect(Collectors.toList());
```